### PR TITLE
Fallback to string if config value is not json

### LINKED ISSUE #1765: Don't require JSON encoding of config values supplied through env.
## Feature Suggestion

When supplying a config value through an env var it currently needs to be JSON-encoded, e.g.

```
APP_CONFIG_app_baseUrl='"http://localhost:3001"'
```

Let's remove that requirement by falling back to assuming the value is a string if it fails `JSON.parse`.

## Possible Implementation

It needs to be implemented in all places where env-vars are read, which is currently [here](https://github.com/spotify/backstage/blob/db5e0408cbbfdac9997fd257458578d698ecb218/packages/config-loader/src/lib/env.ts#L75) and [here](https://github.com/spotify/backstage/blob/db5e0408cbbfdac9997fd257458578d698ecb218/docker/run.sh#L19). If the parsing fails we simply fall back to using the value as-is.

## Context

Multiple requests for this feature on Discord, along with some deployment environments not allowing the JSON-encoding to be passed through at all.

### PR DESCRIPTION
**Pull request by andrewthauer**
## Hey, I just made a Pull Request!

Fixed #1765 - Don't require JSON encoding of config values supplied through env 

image

#### :heavy_check_mark: Checklist

- [x] All tests are passing `yarn test`
- [x] Screenshots attached (for UI changes)
- [x] Relevant documentation updated
- [x] Prettier run on changed files
- [x] Tests added for new functionality
- [x] Regression tests added for bug fixes

### CODE REVIEW

File: docker/run.sh [hunk 1]
```diff
@@ -16,10 +17,10 @@ function inject_config() {
     with_entries(select(.key | startswith("APP_CONFIG_")) | .key |= sub("APP_CONFIG_"; "")) |
     to_entries |
     reduce .[] as $item (
-      {}; setpath($item.key | split("_"); $item.value | fromjson)
+      {}; setpath($item.key | split("_"); $item.value | try fromjson catch $item.value)
     )')"
 
-  >&2 echo "Runtime app config: $config"
+  echo >&2 "Runtime app config: $config"
```
Comment (Rugvip) [inline docker/run.sh:23]: Was this needed by some env or just a nit? :p

File: packages/config-loader/src/lib/env.ts [hunk 2]
```diff
@@ -72,11 +72,16 @@ export function readEnv(env: {
...
           }
           try {
-            const parsedValue = JSON.parse(value);
+            const [err, parsedValue] = safeJsonParse(value);
             if (parsedValue === null) {
               throw new Error('value may not be null');
             }
             obj[part] = parsedValue;
+            if (err) {
+              console.warn(
```
Comment (Rugvip) [inline packages/config-loader/src/lib/env.ts:81]: Tbh I think we should skip the warning. I don't think this new behaviour is something for edge use-cases where you can't get quotes around the value, but rather the standard way that will be recommended through documentation.

We can prolly wait and see if it actually causes trouble for people to not have the warning. Tbh I think the cases for actually wanting a JSON-encoded non-primitive value are rare, and hopefully the failure is somewhat obvious if it happens.

**Review by Rugvip — COMMENTED** [review]
Thanks! :grin:

File: docker/run.sh (hunk 1 above)
Comment (andrewthauer) [reply docker/run.sh:23]: Looks like `shfmt` changed it on me actually. I'll revert back.

**Review by Rugvip — APPROVED** [review]
:shipit:

**Pull request merged**